packages/aws-rfdk/lib/lambdas/nodejs/lib/configure-spot-event-plugin/spot-event-plugin-client.ts

```diff
@@ -94,6 +118,45 @@ export class SpotEventPluginClient {
     }
   }
 
+  public async getCollection(type: CollectionType): Promise<PoolCollections> {
+    console.log(`Getting ${type} collection:`);
+    try {
+      const response = await this.deadlineClient.GetRequest(`/db/settings/collections/${type}s?invalidateCache=true`, {
+        headers: {
+          'Content-Type': 'application/json; charset=utf-8',
+        },
+      });
+      const deadlinePools: PoolCollections = response.data;
+      if (!deadlinePools.Pools || !Array.isArray(deadlinePools.Pools)) {
+        throw new Error(`Failed to receive a ${type} collection. Invalid response: ${JSON.stringify(response.data)}.`);
+      }
+      return deadlinePools;
+    } catch(e) {
+      throw new Error(`Failed to get ${type} collection. Reason: ${(<Error>e).message}`);
+    }
+  }
+
+  public async saveCollection(pools: PoolCollections, type: CollectionType): Promise<boolean> {
+    console.log(`Saving ${type} collection:`);
+    console.log(pools);
+
+    try {
+      await this.deadlineClient.PostRequest(`/db/settings/collections/${type}s/save`, {
+        Pools: pools.Pools,
+        ObsoletePools: pools.ObsoletePools,
+      },
+      {
+        headers: {
+          'Content-Type': 'application/json; charset=utf-8',
+        },
+      });
+      return true;
+    } catch(e) {
+      console.error(`Failed to save ${type} collection. Reason: ${e}`);
```

Comment:
The [code above](https://github.com/aws/aws-rfdk/pull/470/files#diff-153ff117c810c27af1307dcb1f5c3c171197e7d9a54c92fb5ca606c1c0ca9014R135) casts the type of `e` to `Error` and uses the `.message` field. Should we have the same thing here?